### ISSUE TITLE
Allow user overrides when using cactbot remotely

### ISSUE THREAD

**Issue by quisquous** [quisquous/cactbot#56]
It's possible to load cactbot from hosted pages (e.g. https://quisquous.github.io/cactbot/ui/raidboss/raidboss.html), however when doing so there's no way to load local user overrides.

One solution here might be to add all the files from $ACT/OverlayPlugin/cactbot/user/ to the OnDataFilesRead message, so that individual overlays can concat them into the document so scripts and css can be loaded.  If we go this route, then probably not double-loading the local ones would be good too.

Also, minorly, pull counter appears to not work when running remotely (?!).  It appears to not get events and via devtools no breakpoints in log messages ever get hit.  No console errors either.

cc @ezsoftware

**Comment by ezsoftware** [comment]
I was thinking about this as well, I was going to do the changes necessary for the local user file to be correctly injected into the remote source this weekend and submit a pull request.

**Comment by quisquous** [comment]
Thanks! That'd be excellent if you wanted to submit a pull request for this.

**Comment by quisquous** [comment]
One thing that won't work with the current scheme is remote cactbot but local user files.  For security reasons, Chrome isn't keen on letting remote files load local ones.

I wonder if maybe sending the configs needs to send the contents of the config files in this case?

**Comment by ezsoftware** [comment]
Yeah, that is do-able, instead of the payload being a location if its a file, can have the payload be the file contents as a string which can be eval'd in run time.  I'll add this functionality sometime this week when I get a chance and submit a new pull request.

**Comment by quisquous** [comment]
Thanks.  I'll wait for the pull request.

For what it's worth, in an attempt to implement a reliable loading mechanism, I ended up removing the config name being passed to C#.  This lets the load dance be waiting for DOMContentLoaded, C# then pushes the config, and then the plugin initializes itself knowing that dom and options are good (and that events haven't been sent previously).  It makes things a lot simpler when writing an overlay because nothing needs to be done before that config callback.

It's possible to add another step to this dance, but I'm not sure how that would work for plugins that don't need a config (e.g. dps overlays).   I think maybe we could just send everything in the user folder (maybe skipping -example files) and just call it a day for simplicity.

Maybe you have better ideas here?

**Comment by quisquous** [comment]
@ezsoftware, are you still planning to look at this? I have a little bit of time this weekend, and could take a look if you're busy.

**Comment by ezsoftware** [comment]
@quisquous sorry, had a busy weekend, my daughter turned 6 and had friends sleeping over, so was chaos all weekend.  I was going to get to it eventually, but I haven't yet, might this weekend if you haven't by then.

**Comment by quisquous** [comment]
No apologies needed at all!  I just didn't want to step on your toes if you were in the middle of it.  The 1632347 commit seems to do the trick.  I just tested locally with github pages combined with local user files.

**Comment by ezsoftware** [comment]
ok cool, sounds great! I'll download the latest and build locally and test all the remote stuff out for raidboss and oopsy and such in O5s - O8s tomorrow and let you know if I encounter anything strange.

**Comment by ezsoftware** [comment]
@quisquous worked perfectly, no issues, some of the new triggers in O7s surprised me at first, but they were a good improvement, used the pointer files in my bundle that pointed to your github pages and everything went without a hitch for both raidboss and oopsyraidsy.
